assets/js/modules/sign-in-with-google/components/settings/SettingsForm.js

```diff
@@ -30,46 +30,95 @@ import {
 import { MODULES_SIGN_IN_WITH_GOOGLE } from '../../datastore/constants';
 import StoreErrorNotices from '../../../../components/StoreErrorNotices';
 import { Cell, Grid, Row } from '../../../../material-components';
+import {
+	BREAKPOINT_SMALL,
+	BREAKPOINT_TABLET,
+	useBreakpoint,
+} from '../../../../hooks/useBreakpoint';
 
 export default function SettingsForm() {
+	const breakpoint = useBreakpoint();
+	const isSmallScreen =
+		breakpoint === BREAKPOINT_SMALL || breakpoint === BREAKPOINT_TABLET;
+
 	return (
 		<div className="googlesitekit-sign-in-with-google-settings-fields">
 			<StoreErrorNotices
 				moduleSlug="sign-in-with-google"
 				storeName={ MODULES_SIGN_IN_WITH_GOOGLE }
 			/>
-			<Grid>
-				<Row>
-					<Cell size={ 8 }>
-						<Grid>
-							<Row>
-								<Cell size={ 12 }>
-									<ClientIDTextField />
-								</Cell>
-							</Row>
-							<Row>
-								<Cell size={ 4 }>
-									<ButtonTextSelect />
-								</Cell>
-								<Cell size={ 4 }>
-									<ButtonThemeSelect />
-								</Cell>
-								<Cell size={ 4 }>
-									<ButtonShapeSelect />
-								</Cell>
-							</Row>
+			{ isSmallScreen ? (
+				<Grid>
+					<Row>
+						<Cell size={ 8 }>
+							<Grid>
+								<Row>
+									<Cell size={ 12 }>
+										<ClientIDTextField />
+									</Cell>
+								</Row>
+								<Row>
+									<Cell size={ 4 }>
+										<ButtonTextSelect />
+									</Cell>
+									<Cell size={ 4 }>
+										<ButtonThemeSelect />
+									</Cell>
+									<Cell size={ 4 }>
+										<ButtonShapeSelect />
+									</Cell>
+								</Row>
+								<Row>
+									<Cell size={ 12 }>
+										<OneTapToggle />
+									</Cell>
+								</Row>
+							</Grid>
+						</Cell>
+						<Cell size={ 4 }>
+							<Preview />
+						</Cell>
+					</Row>
+				</Grid>
+			) : (
+				<Grid>
+					<Row>
+						<Cell size={ 8 }>
+							<Grid className="googlesitekit-sign-in-with-google-settings-fields__stretch-form">
+								<Row>
+									<Cell size={ 12 }>
+										<ClientIDTextField />
+									</Cell>
+								</Row>
+								<Row>
+									<Cell size={ 4 }>
+										<ButtonTextSelect />
+									</Cell>
+									<Cell size={ 4 }>
+										<ButtonThemeSelect />
+									</Cell>
+									<Cell size={ 4 }>
+										<ButtonShapeSelect />
+									</Cell>
+								</Row>
+							</Grid>
+						</Cell>
+						<Cell
+							size={ 4 }
+							className="googlesitekit-sign-in-with-google-settings-fields__button-preview"
+						>
 							<Row>
 								<Cell size={ 12 }>
-									<OneTapToggle />
+									<Preview />
 								</Cell>
 							</Row>
-						</Grid>
-					</Cell>
-					<Cell size={ 4 }>
-						<Preview />
-					</Cell>
-				</Row>
-			</Grid>
+						</Cell>
+						<Cell size={ 12 }>
+							<OneTapToggle />
+						</Cell>
+					</Row>
+				</Grid>
+			) }
```

Comment:
Let's not use ternaries for big components like this: they get quite hard-to-follow/place.

I'll make the change here, but just a note going forward 🙂

Reply:
Also though: this seems like something we can handle with CSS, this seems a bit complicated/has a lot of duplication. Let's find a CSS-only approach. 🤔

Reply:
What was the reason for this component split based on viewport size? I removed it and it worked just fine with the "large" viewport code.

Reply:
The order of the elements within the grid, in new layout is showing well for desktop, but layout change was needed to re-order one tap toggle on mobile and tablet, so they are above the preview box